### PR TITLE
Horizontal scroll and additional mouse buttons

### PR DESCRIPTION
**Pull request by AdrianKoshka**
Patch talked about in issue #51. Going to mention @vorph1, as I don't know whether they consider this change finished or not.

### CODE REVIEW

File: src/lib/barrier/mouse_types.h [hunk 1]
```diff
@@ -32,10 +32,14 @@ static const ButtonID    kButtonNone   = 0;
...
+static const ButtonID	kButtonExtra0 = 4;
+// browser forwards - mouse5
+static const ButtonID   kButtonExtra1 = 5;
 
 static const ButtonID   kMacButtonRight = 2;
 static const ButtonID   kMacButtonMiddle = 3;
 //@}
 
-static const UInt8      NumButtonIDs  = 5;
+// XXX -- some osx stuff uses this and might expect it to be 5
```
Comment (p12tic) [inline src/lib/barrier/mouse_types.h]: Can we figure out whether this the case or not and explain? For now this seems to have high regression potential.

**Comment by AdrianKoshka** [comment]
Thanks! :D

**Comment by rolandoislas** [comment]
An update on Mojave testing. The original test was performed using a game (League of Legends), which does recognize the button input. However, testing with a browser produces no result for the back/forward buttons.

On Windows 10 the inverse is true. The browser back/forwards works, but the input is not recognized in a game.

**Comment by vorph1** [comment]
NumButtonIDs being 6 shouldn't cause any issues, as for handling the button events it seems it's a client OS issue since the behavior is the same when plugging the device in directly.

**Comment by AdrianKoshka** [comment]
@vorph1 You still interested in this PR?

**Comment by vorph1** [comment]
Sure, I'd like to see this merged. Do you have any other suggestions since as @rolandoislas said the behavior seems to be the same with barrier and mouse plugged in directly?

**Comment by AdrianKoshka** [comment]
Actually, looking back.

> However, testing with a browser produces no result for the back/forward buttons.

> On Windows 10 the inverse is true. The browser back/forwards works, but the input is not recognized in a game

> Edit: I tested the behavior with a mouse physically plugged into the machine running MacOS Mojave and the behavior appears the same. Safari does not support the forward and back buttons and FireFox on MacOS brings up the scroll indicator.

I feel like if I were to merge this, people might complain about having a "half-baked" experience.

**Comment by p12tic** [comment]
I'd argue that given the amount of time the development team has, we should only merge well tested code that don't introduce regressions. It's a pity we don't have unit tests.

**Comment by SavageCore** [comment]
Forward/Back working fine for me with [SensibleSideButtons](https://github.com/archagon/sensible-side-buttons). Is this needed within this PR?

I'm looking for horizontal scroll support, can the two PRs be split?

**Comment by AdrianKoshka** [comment]
I think I'll close this PR due to inactivity. If anyone wants me to re-open it, I'll re-open it under a new PR so it will be built against the new CI system.

**Comment by EbonJaeger** [comment]
I think it would be really nice to get this in. Would it be better to use this PR, or form a new one?

**Comment by AdrianKoshka** [comment]
I'd say new one, so it can be ran against the new CI.

**Pull request closed without merging**